dm/config/task.go

```diff
@@ -239,6 +240,8 @@ type SyncerConfig struct {
 	EnableGTID       bool `yaml:"enable-gtid" toml:"enable-gtid" json:"enable-gtid"`
 	DisableCausality bool `yaml:"disable-detect" toml:"disable-detect" json:"disable-detect"`
 	SafeMode         bool `yaml:"safe-mode" toml:"safe-mode" json:"safe-mode"`
+	// when dump unit can't run consistent dump, enable safe mode until pass exit location of dumping
+	DumpExitLocation *binlog.Location `yaml:"-" toml:"-" json:"-"`
```

Comment:
The name of `*Exit*` may let the user think the dump unit is exit/finished (although it is only used internally), but I have no better name. 😹

Reply:
https://github.com/pingcap/dm/pull/915/commits/16270b591ca3ff2cb491a68759203d19b24a1d97